lib/workers/repository/index.ts

```diff
@@ -39,15 +42,31 @@ export async function renovateRepository(
     logger.debug('Using localDir: ' + localDir);
     config = await initRepo(config);
     addSplit('init');
-    const { branches, branchList, packageFiles } = await extractDependencies(
-      config
+    const { branches, branchList, packageFiles } = await tracer.startActiveSpan(
+      'extractDependencies',
+      async (span) => {
+        const res = await extractDependencies(config);
+        span.end();
+        return res;
+      }
     );
     if (
       GlobalConfig.get('dryRun') !== 'lookup' &&
       GlobalConfig.get('dryRun') !== 'extract'
     ) {
-      await ensureOnboardingPr(config, packageFiles, branches);
-      const res = await updateRepo(config, branches);
+      await tracer.startActiveSpan('ensure onboarding PR', async (span) => {
+        await ensureOnboardingPr(config, packageFiles, branches);
+        span.end();
+      });
+
+      const res = await tracer.startActiveSpan(
+        'update repository',
```

Comment:
Is there any way that we could wrap these into a more generic span tracing function which can be reused in non-OTEL cases too? e.g. to be used to dump stats to our regular logging even when OTEL is not in use?

Reply:
Something like this should work for this scenario:
```typescript
export const instrumentFunction = <A extends any[], R>(
  { name, attributes, ignoreParentSpan, kind }: InstrumentationParameters,
  fn: (...args: A) => R,
  args: A
) => {
  return getTracer().startActiveSpan(
    name,
    { attributes, kind, root: ignoreParentSpan },
    async (span) => {
      // do something
      const result = await fn(...args);
      // do something
      span.end();
      return result;
    }
  );
};

process.exitCode = await instrumentFunction(
  { name: 'run' },
  globalWorker.start,
  []
);
```

Reply:
I have implemented an `instrument` function which would allow this.